### PR TITLE
Add .box support

### PR DESCRIPTION
**Pull request by Har01d**
.box is a domain zone which is routable in both Web2 and Web3. See my.box for details.

### CODE REVIEW

**Comment by Har01d** [comment]
The integration is live: https://3xpl.com/search?q=glo.box

**Review by alexqrid — CHANGES_REQUESTED** [review]
https://3xpl.com/search?q=ohms.box
returns 0x000000000000000000000000000000000000 address for non-existent domains

**Review by alexqrid — APPROVED** [review]

**Comment by Har01d** [comment]
Yes, `ENS Forward Resolution` should be set up at my.box for the domain to be visible with the resolver.

File: Modules/OptimismMainModule.php [hunk 1]
```diff
@@ -27,5 +27,31 @@ function initialize()
...
+
+            $hash = $this->ens_name_to_hash($handle);
+
+            if (is_null($hash) || $hash === '')
+                return null;
+
+            $address = $this->ens_get_data_from_resolver('0xf97aac6c8dbaebcb54ff166d79706e3af7a813c8', $hash, '0x3b3b57de', -40);
+
+            if ($address === '0000000000000000000000000000000000000000') // try to call ownerOf on .box NFT contract
+                $address = $this->ens_get_data_from_resolver('0xBB7B805B257d7C76CA9435B3ffe780355E4C4B17', $hash, '0x6352211e', -40);
```
Comment (alexqrid) [inline Modules/OptimismMainModule.php:49]: @Har01d This resolves domains that didn't setup `ENS Forward Resolution`
As domain's NFT ID is derived from the ENS `namehash` of the domain, we can get the owner of the NFT and the address we get is the resolution of the domain.
I've checked a bunch of domains and didn't found the exception from this rule.

If there's a possibility to set the resolution address different from NFT receiver address, this should be done ENS way, and our first call to the resolver address wouldn't be there.
I believe we should keep this approach to resolve domains without `ENS Forward Resolution`

**Pull request merged**